src/js/components/sidebar.jsx

```diff
@@ -140,6 +164,13 @@ export default class SideBar extends React.Component {
       return;
     }
 
+    // select by click feature
+    let state = this.state;
+    let tab = this.state.tabsById.get(tabId);
+    if (state.selectByClick && !tab.filtered) {
+      tab.selected = true;
+    }
+
     this.setState(Object.assign({}, this.state, { activeTabId: tabId }));
```

Comment:
I know this isn't your code, but all we really need to do here is `this.setState({ activeTabId: tabId })`.